PhotonSieveDevelopment.js

```diff
@@ -24,92 +24,96 @@ class Sieve {
         this.totalPinholes = totalPinholes
         this.totalArea = totalArea
         this.fNumber = fNumber
+        this.rings = []
+    }
+
+    addRing(ring) {
+        this.rings.push(ring);
+        this.totalArea = this.totalArea + ring.openArea;
+        this.totalPinholes = this.totalPinholes + ring.pinholes.length;
+        let sieveEquivalentAperture = Math.sqrt(this.totalArea / Math.PI) * 2;
+        this.fNumber = FD / sieveEquivalentAperture;
     }
 }
 
 class Ring {
-    constructor(zone, centerlineRadius, ringWidth, numOfPinholes, openArea){
+    constructor(zone, centerlineRadius){
         this.zone = zone
         this.centerlineRadius = centerlineRadius
-        this.ringWidth = ringWidth
-        this.numOfPinholes = numOfPinholes
-        this.openArea = openArea
+        this.openArea = 0
+        this.pinholes = []
     } 
+
+    addPinhole(pinhole) {
+        this.pinholes.push(pinhole);
+        this.openArea = this.openArea + pinhole.pinholeArea;
+    }
+
+    // conflict checking function can safely ignore pinholes placed on rings other than current 
+    getConflicts(conflictRadius, conflictAzimuth) {

```

Comment:
Moved this into the Ring object, makes more sense to say "Ring, is this a conflict for you?"